File: cs25backend/src/main/java/edu/dongguk/cs25backend/domain/WarehousingApplication.java
package edu.dongguk.cs25backend.domain;

import jakarta.persistence.*;
import lombok.Builder;
import lombok.Getter;
import lombok.NoArgsConstructor;
import org.hibernate.annotations.DynamicUpdate;

import java.sql.Timestamp;

@Entity
@Getter
@NoArgsConstructor
@DynamicUpdate
@Table(name = "warehousing_application")
public class WarehousingApplication {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "order_id")
    private Long id;

    @Column(name = "supplier_name")
    private String supplierName;

    @Column(name = "created_at")
    private Timestamp createdAt;

    @Column(name = "is_stocked")
    private Boolean isStocked;

    /*--------------------연관 관계 매핑--------------------*/
    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "item_hq_id")
    private ItemHQ itemHQ;


    /*--------------------메서드--------------------*/
    @Builder
    public WarehousingApplication(Long id, String supplierName, Timestamp createdAt, Boolean isStocked, ItemHQ itemHQ) {
        this.id = id;
        this.supplierName = supplierName;
        this.createdAt = createdAt;
        this.isStocked = isStocked;
        this.itemHQ = itemHQ;
    }
}

File: cs25server/src/main/kotlin/edu/dongguk/cs25server/domain/WarehousingApplication.kt
package edu.dongguk.cs25server.domain

import edu.dongguk.cs25server.domain.type.Supplier
import jakarta.persistence.*
import org.hibernate.annotations.DynamicUpdate
import java.sql.Timestamp
import java.time.LocalDate
import java.time.LocalDateTime

@Entity
@DynamicUpdate
@Table(name = "warehousing_application")
class WarehousingApplication(
    @Enumerated(EnumType.STRING)
    private var supplierName: Supplier,

    @Column(name = "created_at")
    private var createdAt: LocalDate = LocalDate.now(),

    @Column(name = "is_stocked")
    private var isStocked: Boolean = false,

    @Column(name = "count")
    private val count: Long,

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "item_hq_id")
    private val itemHQ: ItemHQ? = null
) {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "order_id")
    private val id: Long? = null

    /*--------------------메서드--------------------*/
    fun getCreatedAt():LocalDate {
        return this.createdAt
    }

    fun getIsStocked() = this.isStocked
}